src/python/platforms/fuchsia/undercoat.py

```diff
@@ -46,14 +55,14 @@ class UndercoatError(Exception):
 def undercoat_api_command(*args):
   """Make an API call to the undercoat binary."""
   bundle_dir = environment.get_value('FUCHSIA_RESOURCES_DIR')
-  undercoat_path = os.path.join(bundle_dir, 'undercoat/undercoat')
+  undercoat_path = os.path.join(bundle_dir, 'undercoat', 'undercoat')
   undercoat = new_process.ProcessRunner(undercoat_path, args)
-  # TODO(eep): Is there a more useful way to connect stderr to the
-  # logging system?
-  result = undercoat.run_and_wait(stderr=sys.stderr)
+  error_log = tempfile.TemporaryFile()
+  result = undercoat.run_and_wait(stderr=error_log)
   result.output = result.output.decode('utf-8')
 
   if result.return_code != 0:
+    logs.log_warn(utils.read_from_handle_truncated(error_log, 1024 * 1024))
     raise UndercoatError(
         'Error running undercoat command %s: %s' % (args, result.output))
```

Comment:
Does `result.output` actually make sense here when errors are written to stderr?

Reply:
The clusterfuchsia API is defined as returning an error message via stdout when the exit code is non-zero. stderr, on the other hand, is used for more detailed logging (i.e. log messages from undercoat itself, not directly related to the API) and is echoed here to assist in debugging. Does that address your concern here?

Reply:
I see, it was just a bit confusing that we are logging stderr and then raising an exception with stdout. Perhaps add this in a comment?

Reply:
Good point! I've added clarifying comments.